contracts/Governance.sol

```diff
@@ -0,0 +1,480 @@
+    bool executed;
+    bool canceled;
+  }
+
+  struct Poll {
+    uint256[] voteIds;
+
+    address proposer;
+    string description;
+
+    // vote starts at
+    uint256 startAt;
+    // vote ends at
+    uint256 endAt;
+    uint256 forVotingPower;
+    uint256 againstVotingPower;
+  }
+
+  struct ShareLock {
+    uint256 amount;
+    uint256 votingPower;
+
+    uint256[] votedProposalIds;
+    uint256[] votedPollIds;
+  }
+
+  event paramChange(string key, bytes value);
+  event ProposalCreated(
+    uint256 indexed id,
+
+    address indexed proposer,
+    string description,
+    uint256 startAt,
+    uint256 endAt
+  );
+  event ProposalExecuted(uint256 indexed proposalId, address indexed executor);
+  event ProposalVoted(uint256 indexed proposalId, address indexed voter, bool indexed support, address shareContract, uint256 shareAmount, uint256 votingPower);
+
+  event PollCreated(
+    uint256 indexed id,
+
+    address indexed proposer,
+    string description,
+    uint256 startAt,
+    uint256 endAt
+  );
+  event PollVoted(uint256 indexed proposalId, address indexed voter, bool indexed support, address shareContract, uint256 shareAmount, uint256 votingPower);
+
+  modifier onlyCabinet() {
+    uint256 indexPlus = IBSCValidatorSet(VALIDATOR_CONTRACT_ADDR).currentValidatorSetMap(msg.sender);
+    uint256 numOfCabinets = IBSCValidatorSet(VALIDATOR_CONTRACT_ADDR).numOfCabinets();
+    if (numOfCabinets == 0) {
+      numOfCabinets = 21;
+    }
+
+    require(indexPlus > 0 && indexPlus <= numOfCabinets, "not cabinet");
+    _;
+  }
+
+  function submitProposal(ParamChangeRequest[] memory requests, string memory _description, uint256 voteAt) public onlyCabinet {
+    _paramInit();
+
+    require(voteAt == 0 || voteAt >= block.timestamp, "invalid voteAt");
+    if (voteAt == 0) {
+      voteAt = block.timestamp;
+    }
+
+    uint256 totalRequests = requests.length;
+    require(totalRequests > 0, "empty param change request");
+    uint256 endAt = voteAt + votingPeriod;
+
+    Proposal memory proposal;
+    uint256[] memory requestIds = new uint256[](totalRequests);
+    uint256[] memory voteIds = new uint256[](totalRequests);
+
+    for (uint256 i = 0; i < totalRequests; i++) {
+      paramChangeRequests.push(requests[i]);
+      requestIds[i] = requests.length - 1;
+    }
+
+    proposal.requestIds = requestIds;
+    proposal.voteIds = voteIds;
+    proposal.proposer = msg.sender;
+    proposal.description = _description;
+    proposal.startAt = voteAt;
+    proposal.endAt = endAt;
+
+    proposals.push(proposal);
+
+    emit ProposalCreated(proposals.length - 1, msg.sender, _description, voteAt, endAt);
+  }
+
+  function cancelProposal(uint256 proposalId) external {
+    _paramInit();
+
+    Proposal storage proposal = proposals[proposalId];
+    require(msg.sender == proposal.proposer, "only proposer");
+    ProposalState _state = proposalState(proposalId);
+    require(
+      _state == ProposalState.Pending &&
```

Comment:
`require(!(...))`?

Reply:
only Pending / Active /Timelocked / AwaitingExecution proposals could be canceled